### PR TITLE
Add support for Llama2, Palm, Cohere, Anthropic, Replicate Models - using litellm 

### PR DESCRIPTION
**Pull request by ishaan-jaff**
This PR adds support for models from all the above mentioned providers using https://github.com/BerriAI/litellm/

`ChatLiteLLM()` is integrated into langchain and allows you to call all models using the OpenAI I/O interface 
https://python.langchain.com/docs/integrations/chat/litellm

Here's an example of how to use ChatLiteLLM()
```python
ChatLiteLLM(model="gpt-3.5-turbo")
ChatLiteLLM(model="claude-2", temperature=0.3)
ChatLiteLLM(model="command-nightly")
ChatLiteLLM(model="replicate/llama-2-70b-chat:2c1608e18606fad2812020dc541930f2d0495ce32eee50074220b87300bc16e1")

```

Here's an example of how just using `litellm.completion` works 
```python
from litellm import completion, acompletion

## set ENV variables
# ENV variables can be set in .env file, too. Example in .env.example
os.environ["OPENAI_API_KEY"] = "openai key"
os.environ["COHERE_API_KEY"] = "cohere key"

messages = [{ "content": "Hello, how are you?","role": "user"}]

# openai call
response = completion(model="gpt-3.5-turbo", messages=messages)

# llama2 call
model_name = "replicate/llama-2-70b-chat:2c1608e18606fad2812020dc541930f2d0495ce32eee50074220b87300bc16e1"
response = completion(model_name, messages)

# cohere call
response = completion("command-nightly", messages)

# anthropic call
response = completion(model="claude-instant-1", messages=messages)
```

### CODE REVIEW

File: application/app.py [hunk 1]
```diff
@@ -187,7 +188,7 @@ def complete_stream(question, docsearch, chat_history, api_key, conversation_id)
         )
     else:
         logger.debug("plain OpenAI")
-        llm = ChatOpenAI(openai_api_key=api_key)
+        llm = ChatLiteLLM(openai_api_key=api_key)
```
Comment (ishaan-jaff) [inline application/app.py:191]: ChatLiteLLM allows the same args as ChatOpenAI

**Comment by ishaan-jaff** [comment]
@dartpain can you please take a look at this PR when possible 😊

Happy to add more tests/docs if this initial commit looks good

**Comment by dartpain** [comment]
Hi, thank you for a PR and for considering to submit it with our project, its very important that we can switch between huggingface and normal models, also I see that 
`messages = [{ "content": "Hello, how are you?","role": "user"}]`

structure is different, if you can adapt it please, this would be great
so:

- huggingface compatability
- azure compatability
- messages integrated

Ill convert this PR to a draft for now

**Comment by ishaan-jaff** [comment]
Hi Alex

**Comment by ishaan-jaff** [comment]
Hi @dartpain LiteLLM now has huggingface, azure support and your input remains as messages 

Would love your feedback on this

**Comment by dartpain** [comment]
I had to close this PR as currently it does not match our backend. Think we can integrate LiteLLM very well, it will be very beneficial. But please check out how we connect llms now. In `application/llm` Just add a concrete class, or maybe an adapter for LiteLLM please and I will approve the PR.

**Comment by ishaan-jaff** [comment]
started a new PR

**Pull request closed without merging**